tools/python/maps_generator/generator/stages_declaration.py

```diff
@@ -316,10 +317,13 @@ def apply(self, env: Env, country, **kwargs):
 @country_stage
 @production_only
 class StageMwmDiffs(Stage):
+    """Create diffs for old versions of a single country."""
     def apply(self, env: Env, country, logger, **kwargs):
+        # FIXME: not sure what these were supposed to be, but .mwm and mwm_version is required...
+        # FIXME: this uses the "live" maps directory layout for old versions, should it use maps_build instead?
         data_dir = diffs.DataDir(
-            mwm_name=env.build_name,
-            new_version_dir=env.build_path,
+            country_name=country,
+            new_version_dir=Path(env.build_path, env.mwm_version),
             old_version_root_dir=settings.DATA_ARCHIVE_DIR,
         )
```

Comment:
With the changes here and in [`diffs.py`](https://github.com/organicmaps/organicmaps/pull/8883/files#diff-b9e187e8f86a4a1d6209ceda7d7615036a78d2f1aa9cf24fbf352d4264519243R67-R74), the diff generation works with the current `maps_build` layout:

```
maps_build
├── 2023_05_08__20_38_48
│  ├── 230508
│  │  └── Canada_Yukon_Whitehorse.mwm
│  ...
...
```

and expects `DATA_ARCHIVE_DIR` to look like:
```
DATA_ARCHIVE_DIR
├── 231002
│  └── Canada_Yukon_Whitehorse.mwm
└── 240603
   └── Canada_Yukon_Whitehorse.mwm
```

The diffs are then written to `maps_build/2023_05_08__20_38_48/230508/{old_version}/*.mwmdiff`.

Does that work with how you "release" new maps or is there a better layout to use?

Reply:
@vng also is the proposed dirs structure would be convenient to maintain on the server?
E.g.
```
diffs/240603/231002/Canada_Yukon_Whitehorse.mwmdiff
maps/240603/Canada_Yukon_Whitehorse.mwm
```
to delete a map version from the server one would need to remember to delete from two dirs.
Maybe it could be
```
maps/240603/Canada_Yukon_Whitehorse.mwm
maps/240603/231002-diff/Canada_Yukon_Whitehorse.mwmdiff
```